### PR TITLE
UDP sockets can now use bind/connect in addition to socket/connect

### PR DESCRIPTION
**Pull request by justinboswell**
Depends on: https://github.com/awslabs/aws-crt-builder/pull/43

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/posix/socket.c [hunk 1]
```diff
@@ -526,7 +526,8 @@ int aws_socket_connect(
         return aws_raise_error(AWS_IO_EVENT_LOOP_ALREADY_ASSIGNED);
     }
 
-    if (socket->state != INIT) {
+    /* UDP sockets jump straight to CONNECT_READ once bind is called */
+    if (socket->state != INIT && socket->options.type != AWS_SOCKET_DGRAM) {
```
Comment (JonathanHenson) [inline source/posix/socket.c:530]: You should still check for valid state when udp is used.

Comment (justinboswell) [reply source/posix/socket.c:530]: Good call, fixed.

**Comment by justinboswell** [comment]
> What about windows?

Bah, meant to create a draft :( Not done yet!

File: tests/socket_test.c [hunk 2]
```diff
@@ -253,6 +253,9 @@ static int s_test_socket(
 
     struct aws_socket outgoing;
     ASSERT_SUCCESS(aws_socket_init(&outgoing, allocator, options));
+    if (local && local != endpoint) {
```
Comment (bretambrose) [inline tests/socket_test.c:256]: value comparison instead?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -7,8 +7,8 @@ on:
       - '!master'
 
 env:
-  BUILDER_VERSION: v0.6.0
-  BUILDER_SOURCE: releases
+  BUILDER_VERSION: fixes
```
Comment (bretambrose) [inline .github/workflows/ci.yml:10]: Is this final or would this change once fixes gets merged?

File: tests/socket_test.c (hunk 2 above)
Comment (justinboswell) [reply tests/socket_test.c:256]: I can add that.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (justinboswell) [reply .github/workflows/ci.yml:10]: Once the dependent change is merged, I'll update this to a release.

**Pull request merged**